sdk/inc/azure/core/az_json.h

```diff
@@ -792,6 +792,29 @@ AZ_NODISCARD az_result az_json_reader_next_token(az_json_reader* ref_json_reader
  */
 AZ_NODISCARD az_result az_json_reader_skip_children(az_json_reader* ref_json_reader);
 
+/**
+ * @brief Gets the JSON token's string after unescaping it, if required.
```

Comment:
The input parameter is not a `json_token` but rather `az_span`, so consider updating this comment

```suggestion
 * @brief Unescapes the JSON string within the provided #az_span.
```